### ISSUE TITLE
Multiline subprocess mode not working

### ISSUE THREAD

**Issue by GuillaumeLeclerc** [xonsh/xonsh#893]
Hello,

When I type: 

```
$ echo "1 \
2"
```

I get an error:

`xonsh: subprocess mode: command not found: 2"`

**Comment by scopatz** [comment]
This seems like a bug with trailing backslashes in general.

**Comment by scopatz** [comment]
Thanks for reporting @GuillaumeLeclerc!

**Comment by a-da** [comment]
reproduced with docker:

    $ docker run -it python:3.5 bash
    $ pip install xonsh
    $ xonsh
    ...
    1, 2, or 3 [default: 2]? 
    <user>@a38d91d222f9 / # echo "a \
    ....................... b"
    ....................... 
    "a \
    xonsh: subprocess mode: command not found: b"
    Did you mean one of the following?
        w:   Command (/usr/bin/w)
        c[:   Command (/usr/bin/[)
        bg:  Alias
        id:  Command (/usr/bin/id)
        dd:  Command (/bin/dd)
    <user>@a38d91d222f9 / # exit

**Comment by scopatz** [comment]
The problem seems to be that the first line is getting autowrapped in subproc-mode too soon.  Note that if you have explicitly put things in subproc mode to start, everything works out just fine:

```console
scopatz@artemis ~/xonsh master $ ![echo "a \
`·.,¸,.·*¯`·.,¸,.·*¯`·.,¸,.·*¯`· b"]
`·.,¸,.·*¯`·.,¸,.·*¯`·.,¸,.·*¯`· 
a b
scopatz@artemis ~/xonsh master $ $XONSH_DEBUG = 1
scopatz@artemis ~/xonsh master $ echo "a \
<xonsh-code>:0:0 - echo "a \
<xonsh-code>:0:0 + ![echo "a \]
<xonsh-code>:1:5 - echo "a \
<xonsh-code>:1:5 + ![echo "a \]
`·.,¸,.·*¯`·.,¸,.·*¯`·.,¸,.·*¯`· b"
`·.,¸,.·*¯`·.,¸,.·*¯`·.,¸,.·*¯`· 
<xonsh-code>:1:5 - echo "a \
<xonsh-code>:1:5 + ![echo "a \]
<xonsh-code>:2:1 - b"
<xonsh-code>:2:1 + ![b"]
"a \
xonsh: To log full traceback to a file set: $XONSH_TRACEBACK_LOGFILE = <filename>
Traceback (most recent call last):
  File "/home/scopatz/.local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 15825, in run_subproc
    proc = cls(aliased_cmd,
  File "/home/scopatz/miniconda3/lib/python3.5/subprocess.py", line 947, in __init__
    restore_signals, start_new_session)
  File "/home/scopatz/miniconda3/lib/python3.5/subprocess.py", line 1551, in _execute_child
    raise child_exception_type(errno_num, err_msg)
FileNotFoundError: [Errno 2] No such file or directory: 'b"'

During handling of the above exception, another exception occurred:
```

The AST transformer is probably not aware of `\\n` and the tokenizer/lexer/parser is treating them as an actual newline. One of these assumptions should change and I am not sure which would be easier.

**Comment by scopatz** [comment]
Potential fix in #2194

**Comment by scopatz** [comment]
This has been fixed in #2194. Please feel free to reopen if needed.
